lib/surface/formatter.ex

```diff
@@ -376,13 +376,22 @@ defmodule Surface.Formatter do
   """
   @spec format_string!(String.t(), list(option)) :: String.t()
   def format_string!(string, opts \\ []) do
+    trailing_whitespace =
+      case Regex.run(~r/\s+$/, string) do
```

Comment:
I think we want to check if there's a trailing newline, not just any trailing whitespace.

In the current form, this:

```elixir
~F{<div />    }
```

would be formatted as this:

```elixir
~F{<div />
}
```

But if we change the regex as such:

```suggestion
    trailing_newline =
      case Regex.run(~r/\n\s+$/, string) do
```

then it will be formatted as

```elixir
~F{<div />}
```

which I think makes more sense. What do you think?

Reply:
Totally -- I'll add a test for this as well.